src/identify/index.js

```diff
@@ -118,8 +118,8 @@ function updateSelf (peerSelf, observedAddr) {
   }
 
   for (var i = 0; i < peerSelf.previousObservedAddrs.length; i++) {
-    if (peerSelf.previousObservedAddrs[i].toString() === omh.toString()) {
-      peerSelf.previousObserveredAddrs.splice(i, 1)
+    if (multiaddr(peerSelf.previousObservedAddrs[i]).toString() === omh.toString()) {
```

Comment:
Good catch and yes, that is indeed the solution, that was my intent but it seems that I missed it. Thank you again :) 

Apologies for the delay in my response, I've been traveling for the last day and half to Battlemesh.